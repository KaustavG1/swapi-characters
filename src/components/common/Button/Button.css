.button-default {
  border-radius: 5px;
  border: 1px solid #1f1f1f;
  cursor: pointer;
  background-color: #fffeff;
}
