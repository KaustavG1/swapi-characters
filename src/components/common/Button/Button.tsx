import "./Button.css";

export interface ButtonProps {
  icon?: string;
  text: string;
  disabled?: boolean;
  className?: string;
}

function Button({ icon, text, className, disabled }: ButtonProps) {
  return (
    <div>
      <button className={className} disabled={disabled}>
        <span>{icon}</span>
        <span>{text}</span>
      </button>
    </div>
  );
}

export default Button;
