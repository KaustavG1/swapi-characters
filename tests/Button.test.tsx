import { it, expect, vi } from "vitest";
import { fireEvent, render, screen } from "@testing-library/react";
import Button from "../src/components/common/Button/Button";

describe("Button component", () => {
  it("renders 'Button' component", () => {
    const onClick = vi.fn();
    const { container } = render(
      <Button text="Test button" onClick={onClick} />
    );
    const button = screen.getByText(/Test button/i);
    expect(button).toBeInTheDocument();
    fireEvent.click(button);
    expect(onClick).toHaveBeenCalledOnce();
    expect(container).toMatchSnapshot();
  });

  it("renders disabled 'Button' component", () => {
    const { container } = render(<Button text="Test Button" disabled={true} />);
    const button = screen.getByText(/Test Button/i).parentNode;
    expect(button).toBeInTheDocument();
    expect(button).toBeDisabled();
    expect(container).toMatchSnapshot();
  });
});
