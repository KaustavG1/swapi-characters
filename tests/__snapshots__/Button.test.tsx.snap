// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > renders 'Button' component 1`] = `
<div>
  <div>
    <button
      class="undefined button-default"
    >
      <span />
      <span>
        Test button
      </span>
    </button>
  </div>
</div>
`;

exports[`Button component > renders disabled 'Button' component 1`] = `
<div>
  <div>
    <button
      class="undefined button-default"
      disabled=""
    >
      <span />
      <span>
        Test Button
      </span>
    </button>
  </div>
</div>
`;
